packages/console/src/containers/AppContent/index.tsx

```diff
@@ -29,16 +30,22 @@ const AppContent = () => {
 
   const location = useLocation();
   const navigate = useNavigate();
+  const [searchParameters] = useSearchParams();
   const { firstItem } = useSidebarMenuItems();
   const scrollableContent = useRef<HTMLDivElement>(null);
   const { scrollTop } = useScroll(scrollableContent.current);
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
 
   useEffect(() => {
     if (!isAuthenticated && !isLogtoLoading) {
-      void signIn(new URL(href, window.location.origin).toString());
+      const signUpFirst = searchParameters.get(queryParameterKeys.signUp);
+
+      void signIn(
+        new URL(href, window.location.origin).toString(),
+        conditional(signUpFirst === 'true' && 'signUp')
+      );
     }
-  }, [href, isAuthenticated, isLogtoLoading, signIn]);
+  }, [href, isAuthenticated, isLogtoLoading, searchParameters, signIn]);
```

Comment:
a direct access should be sign-in? maybe we only need to set to sign-up when user comes from elsewhere like website and product hunt

Reply:
but for now we cannot configure to use sign-up as the first screen by default. keep as it is looks ok

Reply:
The current use case only supports the Logto Official WebSite's button, assume all lands at the register page.

Reply:
yes, so keep as it is looks ok now